runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/Engine.java

```diff
@@ -196,24 +220,23 @@ public long counter(
         return counter.applyAsLong(name);
     }
 
-    public Future<Void> start()
+    public Future<Void> start() throws Exception
     {
         for (AgentRunner runner : runners)
         {
             AgentRunner.startOnThread(runner, Thread::new);
         }
-
-        this.configureRef = commonPool().submit(configure);
-
-        return configureRef;
+        commonPool().submit(watcherTask);
+        watcherTask.onURLDiscovered(rootConfigURL);
+        return CompletableFuture.completedFuture(null);
     }
 
     @Override
     public void close() throws Exception
     {
         final List<Throwable> errors = new ArrayList<>();
 
-        configureRef.cancel(true);
+        watcherTask.interrupt();
```

Comment:
See reply comment in PR conversation about considering `watcherTask.close()` approach here.